app/gui2/src/components/visualizations/ScatterplotVisualization.vue

```diff
@@ -132,35 +144,71 @@ const SHAPE_TO_SYMBOL: Record<string, d3.SymbolType> = {
   triangle: d3.symbolTriangle,
 }
 
-const SCALE_TO_D3_SCALE: Record<ScaleType, () => d3.ScaleContinuousNumeric<number, number>> = {
+const SCALE_TO_D3_SCALE: any = {
   [ScaleType.Linear]: () => d3.scaleLinear(),
   [ScaleType.Logarithmic]: () => d3.scaleLog(),
+  [ScaleType.Time]: () => d3.scaleTime(),
+}
+
+const createDateTime = (x: DateObj) => {
+  const dateTime = new Date()
+  x.day ? dateTime.setDate(x.day) : dateTime
+  x.month ? dateTime.setMonth(x.month) : dateTime
+  x.year ? dateTime.setFullYear(x.year) : dateTime
+  x.hour ? dateTime.setHours(x.hour) : dateTime
+  x.minute ? dateTime.setMinutes(x.minute) : dateTime
+  x.second ? dateTime.setSeconds(x.second) : dateTime
+  return dateTime
 }
 
 const data = computed<Data>(() => {
   let rawData = props.data
   const unfilteredData =
     Array.isArray(rawData) ? rawData.map((y, index) => ({ x: index, y })) : rawData.data ?? []
-  const data: Point[] = unfilteredData.filter(
-    (point) =>
-      typeof point.x === 'number' &&
-      !Number.isNaN(point.x) &&
-      typeof point.y === 'number' &&
-      !Number.isNaN(point.y),
-  )
+  let data: Point[]
+  // eslint-disable-next-line camelcase
+  const isTimeSeries: boolean =
+    'x_value_type' in rawData ?
+      rawData.x_value_type === 'Time' ||
+      rawData.x_value_type === 'Date' ||
+      rawData.x_value_type === 'Date_Time (with timezone)'
```

Comment:
Why do we receive that full readable name as type? Shouldn't it be just `Date_Time`? Leaving it like this is brittle, as it will break as soon as the human readable representation is changed for any reason.